Sources/Workspace/Workspace.swift

```diff
@@ -3589,6 +3645,28 @@ extension CheckoutState {
     }
 }
 
+extension Workspace {
```

Comment:
potentially this can be an extension Workspace.ManagedDependency?

e.g. (have not tried to compile)


```
extension Workspace.ManagedDependency {
     fileprivate func getCustomFileSystem(packageContainerProvider: PackageContainerProvider?, observabilityScope: ObservabilityScope) throws -> FileSystem? {
         guard case .custom(_, _) = self else {
             observabilityScope.emit(error: "invalid managed dependency state for custom dependency: \(state)")
             return nil
         }

         let container = try temp_await { packageContainerProvider(for: package, skipUpdate: true, observabilityScope: observabilityScope, on: .sharedConcurrent, completion: $0) }
         guard let customContainer = container as? CustomPackageContainer else {
             observabilityScope.emit(error: "invalid custom dependency container: \(container)")
             return nil
         }

         return try customContainer.getFileSystem()
     }
 }


```

Reply:
I'd rather not involve managed dependencies more :)